addons/website_mail_channel/controllers/main.py

```diff
@@ -14,47 +14,35 @@ class MailGroup(http.Controller):
-            end_date = datetime.datetime.strptime(group['__domain'][1][2], tools.DEFAULT_SERVER_DATETIME_FORMAT).date()
-            group['date_begin'] = '%s' % datetime.date.strftime(begin_date, tools.DEFAULT_SERVER_DATE_FORMAT)
-            group['date_end'] = '%s' % datetime.date.strftime(end_date, tools.DEFAULT_SERVER_DATE_FORMAT)
+            group['date_begin'] = '%s' % Date.to_string(Datetime.from_string(group['__domain'][0][2]).date())
+            group['date_end'] = '%s' % Date.to_string(Datetime.from_string(group['__domain'][1][2]).date())
         return groups
 
     @http.route("/groups", type='http', auth="public", website=True)
     def view(self, **post):
-        cr, uid, context = request.cr, request.uid, request.context
-        group_obj = request.registry.get('mail.channel')
-        mail_message_obj = request.registry.get('mail.message')
-        group_ids = group_obj.search(cr, uid, [('alias_id', '!=', False), ('alias_id.alias_name', '!=', False)], context=context)
-        groups = group_obj.browse(cr, uid, group_ids, context)
+        groups = request.env['mail.channel'].search([('alias_id', '!=', False), ('alias_id.alias_name', '!=', False)])
 
         # compute statistics
-        month_date = datetime.datetime.today() - relativedelta.relativedelta(months=1)
-        result = mail_message_obj.read_group(
-            cr, SUPERUSER_ID,
-            [('model', '=', 'mail.channel'), ('date', '>=', month_date.strftime(tools.DEFAULT_SERVER_DATETIME_FORMAT))],
-            [], ['res_id'],
-            context=context)
-        result = dict([(x['res_id'], x['res_id_count']) for x in result])
-
-        group_data = dict()
-        for group in groups:
-            group_data[group.id] = {'monthly_message_nbr': result.get(group.id, 0)}
+        month_date = Datetime.from_string(Datetime.now()) - relativedelta.relativedelta(months=1)
+        messages = request.env['mail.message'].read_group(
+            [('model', '=', 'mail.channel'), ('date', '>=', Datetime.to_string(month_date))],
+            [], ['res_id'])
+        message_data = {message['res_id']: message['res_id_count'] for message in messages}
+
+        group_data = {group.id: {'monthly_message_nbr': message_data.get(group.id, 0)} for group in groups}
 
-        values = {'groups': groups, 'group_data': group_data}
-        return request.website.render('website_mail_channel.mail_channels', values)
+        return request.website.render('website_mail_channel.mail_channels', {'groups': groups, 'group_data': group_data})
 
     @http.route(["/groups/is_member"], type='json', auth="public", website=True)
     def is_member(self, channel_id=0, **kw):
         """ Determine if the current user is member of the given channel_id
             :param channel_id : the channel_id to check
         """
-        current_user_id = request.uid
+        current_user_id = request.env.uid
```

Comment:
can you refactor line:51 ? I think, no need to browse current user record. use request.env.user.partner_id